tests/rptest/tests/partition_balancer_test.py

```diff
@@ -475,16 +475,26 @@ def test_full_nodes(self):
         self.topic = TopicSpec(partition_count=32)
         self.client().create_topic(self.topic)
 
-        # produce around 2GB of data, this should be enough to fill node disks
-        # to a bit more than 70% usage on average
-        msg_count = 19_000
-        producer = KgoVerifierProducer(self.test_context,
-                                       self.redpanda,
-                                       self.topic,
-                                       msg_size=102_400,
-                                       msg_count=msg_count)
-        producer.start(clean=False)
-        producer.wait_for_acks(msg_count, timeout_sec=120, backoff_sec=5)
+        def get_avg_disk_usage():
+            return sum([
+                self.redpanda.get_node_disk_usage(n)
+                for n in self.redpanda.nodes
+            ]) / len(self.redpanda.nodes) / disk_size
+
+        msg_size = 102_400
+        produce_batch_size = ceil(disk_size / msg_size / 20)
+        while get_avg_disk_usage() < 0.7:
+            producer = KgoVerifierProducer(self.test_context,
```

Comment:
Hmm I wonder if it is really okay to create a lot of producer services like this. Won't this lead to the explosion of `KgoVerifierProducer` directories in the test output? (In the latest test results there are none, which is in itself strange)

Reply:
Yeah, I think I should create just one producer and use it